silk/tests/openthread/ot_test_traffic_router_sleepy_device.py

```diff
@@ -0,0 +1,145 @@
+                                           channel=random.randint(11, 25),
+                                           fabric_id="{0:06x}dead".format(random.randint(0, 0xffffff)))
+
+        cls.thread_sniffer_init(cls.network_data.channel)
+
+    @classmethod
+    @testcase.teardown_class_decorator
+    def tearDownClass(cls: 'TestTrafficRouterSleepyEndDevice'):
+        for device in cls.device_list:
+            device.tear_down()
+
+    @testcase.setup_decorator
+    def setUp(self):
+        pass
+
+    @testcase.teardown_decorator
+    def tearDown(self):
+        pass
+
+    @testcase.test_method_decorator
+    def test01_Pairing(self):
+        # allowlisting between leader and sleepy end device
+        self.sleepy_end_device.allowlist_node(self.router)
+        self.router.allowlist_node(self.sleepy_end_device)
+
+        self.router.form(self.network_data, "router")
+        self.router.permit_join(60)
+        self.wait_for_completion(self.device_list)
+
+        self.logger.info(self.router.ip6_lla)
+        self.logger.info(self.router.ip6_thread_ula)
+
+        self.network_data.xpanid = self.router.xpanid
+        self.network_data.panid = self.router.panid
+
+        self.sleepy_end_device.join(self.network_data, "sleepy-end-device")
+        self.wait_for_completion(self.device_list)
+
+    @testcase.test_method_decorator
+    def test02_NCP_State(self):
+        self.assertEqual(self.router.getprop(wpan.WPAN_STATE), wpan.STATE_ASSOCIATED)
+        self.assertEqual(self.router.getprop(wpan.WPAN_NAME), self.sleepy_end_device.getprop(wpan.WPAN_NAME))
+        self.assertEqual(self.router.xpanid, self.sleepy_end_device.xpanid)
+        self.assertEqual(self.router.panid, self.sleepy_end_device.panid)
+
+    @testcase.test_method_decorator
+    def test03_Transmit_Receive(self):
+
+        class AddressType(enum.Enum):
+            LLA = 0
+            MLA = 1
+
+        addresses = [
+            (self.router, self.sleepy_end_device, AddressType.LLA, AddressType.LLA),
+            (self.router, self.sleepy_end_device, AddressType.LLA, AddressType.MLA),
+            (self.router, self.sleepy_end_device, AddressType.MLA, AddressType.MLA),
+            (self.sleepy_end_device, self.router, AddressType.LLA, AddressType.LLA),
+            (self.sleepy_end_device, self.router, AddressType.LLA, AddressType.MLA),
+            (self.sleepy_end_device, self.router, AddressType.MLA, AddressType.MLA)
+        ]
+
+        for poll_interval in [10, 100, 300]:
+            self.sleepy_end_device.setprop(wpan.WPAN_POLL_INTERVAL, str(poll_interval))
+            interval = self.sleepy_end_device.getprop(wpan.WPAN_POLL_INTERVAL)
+            self.assertEqual(int(interval), poll_interval)
+
+            for i, (src, dst, src_type, dst_type) in enumerate(addresses):
+                port = random.randint(10000 + i * 100, 10099 + i * 100)
+                timeout = 5
+                delay = 1
```

Comment:
Do these two variables need to be redefined for each loop cycle? Potentially could be moved outside of the loop.

Reply:
Ack